### PR TITLE
Tema development

### PR DESCRIPTION
**Pull request by misstbn101**
successfull MVP activity feed, simply lists created conversations

### CODE REVIEW

File: src/main/java/codeu/controller/LoginServlet.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ void setUserStore(UserStore userStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
```
Comment (julieefreeman) [inline src/main/java/codeu/controller/LoginServlet.java]: would revert this change since you're just adding a space.

File: src/main/java/codeu/controller/ConversationServlet.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ void setUserStore(UserStore userStore) {
    * for use by the test framework or the servlet's init() function.
    */
   void setConversationStore(ConversationStore conversationStore) {
-    this.conversationStore = conversationStore;
```
Comment (julieefreeman) [inline src/main/java/codeu/controller/ConversationServlet.java]: would revert this change since you're just adding a space.

File: src/main/webapp/WEB-INF/view/ActivityFeed.jsp [hunk 3]
```diff
@@ -13,45 +13,58 @@
...
-    <a href="/conversations">Conversations</a>
-    <% if(request.getSession().getAttribute("user") != null){ %>
-      <a>Hello <%= request.getSession().getAttribute("user") %>!</a>
-    <% } else{ %>
-      <a href="/activityfeed">Login</a>
-    <% } %>
-    <a href="/about.jsp">About</a>
+  <a id="navTitle" href="/">CodeU Chat App</a> <a href="/conversations">Conversations</a>
+  <% if (request.getSession().getAttribute("user") != null) { %>
+  <a>Hello <%=request.getSession().getAttribute("user")%>!</a>
```
Comment (julieefreeman) [inline src/main/webapp/WEB-INF/view/ActivityFeed.jsp]: Want to maintain indents like the code used to here. So that you can tell what logic is within the if/else statements.

File: src/main/webapp/WEB-INF/view/ActivityFeed.jsp [hunk 4]
```diff
@@ -13,45 +13,58 @@
...
-      <label for="password">Password: </label>
-      <br/>
-      <input type="password" name="password" id="password">
-      <br/><br/>
-      <button type="submit">Login</button>
-    </form>
-
-    <p>New users can register <a href="/register">here</a>.</p>
+
+	<%
```
Comment (julieefreeman) [inline src/main/webapp/WEB-INF/view/ActivityFeed.jsp]: sorry to be so nit-picky but can you maintain the previous styling? 

<% if(request.getAttribute("error") != null){ %>

rather than <% on it's own lines. Same for throughout this file.

File: src/main/webapp/WEB-INF/view/ActivityFeed.jsp [hunk 5]
```diff
@@ -13,45 +13,58 @@
...
+
+	<h2>This where you see what the world is up to!</h2>
+
+	<%
+	List<Conversation> conversations = (List<Conversation>) request.getAttribute("conversations");
+	%>
+	<ul class="mdl-list">
+	<%
+	  for (Conversation conversation : conversations) {
+	  	/* String author = UserStore.getInstance()
```
Comment (julieefreeman) [inline src/main/webapp/WEB-INF/view/ActivityFeed.jsp]: remove commented out code

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 6]
```diff
@@ -55,7 +55,8 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
       <a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
- 		<a href="/profile">Profile</a>
+ 	<a href="/profile">Profile</a>
```
Comment (julieefreeman) [inline src/main/webapp/WEB-INF/view/chat.jsp]: Should these lines have the same indent as the About line?

File: src/test/java/codeu/controller/ActivityFeedServletTest.java [hunk 7]
```diff
@@ -21,17 +21,17 @@
...
+  activityfeedServlet = new ActivityFeedServlet(); 
+  mockRequest = Mockito.mock(HttpServletRequest.class);
+  mockResponse = Mockito.mock(HttpServletResponse.class);
+  mockOutputStream = Mockito.mock(ServletOutputStream.class);
+  Mockito.when(mockResponse.getOutputStream())
+        .thenReturn(mockOutputStream);
   }
-	
-  @Test
+  /*@Test
```
Comment (julieefreeman) [inline src/test/java/codeu/controller/ActivityFeedServletTest.java]: Why comment out this test?

File: src/test/java/codeu/controller/ActivityFeedServletTest.java [hunk 8]
```diff
@@ -21,17 +21,17 @@
 
   @Before
   public void setup() throws IOException {
-    activityfeedServlet = new ActivityFeedServlet(); 
-    mockRequest = Mockito.mock(HttpServletRequest.class);
-    mockResponse = Mockito.mock(HttpServletResponse.class);
-    mockOutputStream = Mockito.mock(ServletOutputStream.class);
-    Mockito.when(mockResponse.getOutputStream())
-    .thenReturn(mockOutputStream);
+  activityfeedServlet = new ActivityFeedServlet();
```
Comment (julieefreeman) [inline src/test/java/codeu/controller/ActivityFeedServletTest.java]: Revert all these changes? I think the indents were right before

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 9]
```diff
@@ -1,14 +1,63 @@
...
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-  	response.getOutputStream().println("This is the Activity Feed");	
-  }
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.UserStore;
+
+public class ActivityFeedServlet extends HttpServlet {
```
Comment (julieefreeman) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: It looks like throughout this file you've indented 4 spaces instead of 2.

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 10]
```diff
@@ -1,14 +1,63 @@
...
+		this.conversationStore = conversationStore;
+	}
+
+	@Override
+	public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+		List<Conversation> conversations = conversationStore.getAllConversations();
+		request.setAttribute("conversations", conversations);
+		
+		/*UUID conversationId = ((Conversation) conversations).getId(); */
```
Comment (julieefreeman) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: Remove commented out code

File: src/main/java/codeu/controller/LoginServlet.java (hunk 1 above)
Comment (misstbn101) [reply src/main/java/codeu/controller/LoginServlet.java]: Sorry, I'm not totally sure how I would revert that change back, how did I add a space?

File: src/main/webapp/WEB-INF/view/ActivityFeed.jsp (hunk 3 above)
Comment (misstbn101) [reply src/main/webapp/WEB-INF/view/ActivityFeed.jsp]: I think I fixed

File: src/test/java/codeu/controller/ActivityFeedServletTest.java (hunk 7 above)
Comment (misstbn101) [reply src/test/java/codeu/controller/ActivityFeedServletTest.java]: I commented out the test because it was an old test for when I had the prototype

File: src/test/java/codeu/controller/ActivityFeedServletTest.java (hunk 8 above)
Comment (misstbn101) [reply src/test/java/codeu/controller/ActivityFeedServletTest.java]: How do I revert a change

File: src/main/java/codeu/controller/ConversationServlet.java (hunk 2 above)
Comment (misstbn101) [reply src/main/java/codeu/controller/ConversationServlet.java]: How do I revert a change?

**Comment by misstbn101** [comment]
I addressed the reviews I was clear on and then replied with questions to the ones I was confused on. My next task will be to make some tests.

File: src/main/java/codeu/controller/ConversationServlet.java (hunk 2 above)
Comment (julieefreeman) [reply src/main/java/codeu/controller/ConversationServlet.java]: Just remove the space.

File: src/main/java/codeu/controller/ActivityFeedServlet.java (hunk 9 above)
Comment (julieefreeman) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: You'll need to do this everywhere. 
e.g 
public void init() throws ServletException {
 		super.init();
                ....

should be 

public void init() throws ServletException {
  super.init();
  ....

File: src/main/webapp/WEB-INF/view/ActivityFeed.jsp (hunk 3 above)
Comment (julieefreeman) [reply src/main/webapp/WEB-INF/view/ActivityFeed.jsp]: `
<% if (request.getSession().getAttribute("user") != null) { %>
  <a>Hello <%=request.getSession().getAttribute("user")%>!</a>
<% } else { %>
`

Comment (julieefreeman) [reply src/main/webapp/WEB-INF/view/ActivityFeed.jsp]: The <a> should be indented more than the if statment, same for the code between the {} of the else

File: src/main/webapp/WEB-INF/view/login.jsp [hunk 11]
```diff
@@ -30,7 +30,8 @@
       <a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
-		<a href="/profile">Profile</a>
+	<a href="/profile">Profile</a>
```
Comment (julieefreeman) [inline src/main/webapp/WEB-INF/view/login.jsp]: wrong indents

File: src/main/webapp/about.jsp [hunk 12]
```diff
@@ -30,7 +30,8 @@
       <a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
-		<a href="/profile.jsp">Profile</a>
+	<a href="/profile.jsp">Profile</a>
```
Comment (julieefreeman) [inline src/main/webapp/about.jsp]: wrong indents here too i think

File: src/test/java/codeu/controller/ActivityFeedServletTest.java (hunk 7 above)
Comment (julieefreeman) [reply src/test/java/codeu/controller/ActivityFeedServletTest.java]: Okay. Should update the test to work for the code as it is written now, otherwise remove it.

File: src/main/webapp/WEB-INF/view/ActivityFeed.jsp [hunk 13]
```diff
@@ -26,31 +26,41 @@
...
+    <a href="/about.jsp">About</a>
+    <a href="/users/<%= request.getSession().getAttribute("user") %>">Profile</a>
+    <a href="/profile">Profile</a>
+    <a href="/activityfeed">Activity Feed</a>
   </nav>
 
 
   <div id="container">
 
-	<%if (request.getAttribute("error") != null) { %>
```
Comment (julieefreeman) [inline src/main/webapp/WEB-INF/view/ActivityFeed.jsp]: Why remove the error handling?
